service/mcs/persistent/src/main/java/eu/europeana/cloud/service/mcs/persistent/cassandra/CassandraDataSetDAO.java

```diff
@@ -681,46 +685,83 @@ private Representation mapRowToRepresentationStub(Row row){
+    public void removeDataSetsRevision(String providerId, String datasetId, Revision revision, String
             representationName, String cloudId) {
-        String providerDatasetId = createProviderDataSetId(providerId,datasetId);
-        BoundStatement boundStatement = removeDataSetsRevision.bind(providerDatasetId, revisionId,
+        BoundStatement boundStatement = removeDataSetsRevision.bind(providerId, datasetId, revision.getRevisionProviderId(), revision.getRevisionName(), revision.getCreationTimeStamp(),
                 representationName, cloudId);
         ResultSet rs = connectionProvider.getSession().execute(boundStatement);
         QueryTracer.logConsistencyLevel(boundStatement, rs);
     }
 
-    public List<String> getDataSetsRevisionWithPagination(String providerId, String datasetId, String revisionId, String representationName, String startFromCloudId, int limit){
-        String providerDatasetId = createProviderDataSetId(providerId,datasetId);
-        BoundStatement boundStatement = getDataSetsRevisionWithPagination.bind(providerDatasetId,revisionId,representationName,startFromCloudId,limit);
+//    public List<String> getDataSetsRevisionWithPagination(String providerId, String datasetId, String revisionId, String representationName, String startFromCloudId, int limit){
+//        String providerDatasetId = createProviderDataSetId(providerId,datasetId);
+//        BoundStatement boundStatement = getDataSetsRevisionWithPagination.bind(providerDatasetId,revisionId,representationName,startFromCloudId,limit);
+//        ResultSet rs = connectionProvider.getSession().execute(boundStatement);
+//        QueryTracer.logConsistencyLevel(boundStatement, rs);
+//        List<String> cloudIds = new ArrayList<>();
+//        for (Row r : rs){
+//            String cloudId = r.getString("cloud_Id");
+//            cloudIds.add(cloudId);
+//        }
+//        return cloudIds;
+//    }
+
+//    public List<String> getDataSetsRevision(String providerId, String datasetId, String revisionId, String representationName, int limit){
+//        String providerDatasetId = createProviderDataSetId(providerId,datasetId);
+//        BoundStatement boundStatement = getDataSetsRevision.bind(providerDatasetId,revisionId,representationName, limit);
+//        ResultSet rs = connectionProvider.getSession().execute(boundStatement);
+//        QueryTracer.logConsistencyLevel(boundStatement, rs);
+//        List<String> cloudIds = new ArrayList<>();
+//        for (Row r : rs){
+//            String cloudId = r.getString("cloud_Id");
+//            cloudIds.add(cloudId);
+//        }
+//        return cloudIds;
+//    }
+
+    public List<Properties> getDataSetsRevisions(String providerId, String dataSetId, String revisionProviderId, String revisionName, Date revisionTimestamp, String representationName, String nextToken, int limit) {
+        List<Properties> result = new ArrayList<>(limit);
```

Comment:
That is not possible because one of the object on the list is not of CloudTagsResponse class. It is the one which contains the string for the next slice generated by the cassandra driver.